File: backend/app/utils/regos_integration_token.py
import re

_GATEWAY_OUT_TOKEN = re.compile(r"/gateway/out/([^/?#\s]+)", re.IGNORECASE)


def extract_integration_token(value: str) -> str:
    trimmed = value.strip()
    if not trimmed:
        return ""
    match = _GATEWAY_OUT_TOKEN.search(trimmed)
    if match:
        return match.group(1).strip()
    return trimmed

File: frontend/src/lib/regos-integration-token.ts
const GATEWAY_OUT_TOKEN = /\/gateway\/out\/([^/?#\s]+)/i;

export function extractRegosIntegrationToken(input: string): string {
  const trimmed = input.trim();
  if (!trimmed) return "";
  const match = GATEWAY_OUT_TOKEN.exec(trimmed);
  return match ? match[1].trim() : trimmed;
}
